python/cudaq/kernel/ast_bridge.py

```diff
@@ -231,26 +231,68 @@ def getIntegerAttr(self, type, value):
         """
         return IntegerAttr.get(type, value)
 
-    def getFloatType(self):
+    def getFloatType(self, width=64):
         """
-        Return an MLIR float type (double precision).
+        Return an MLIR float type (single or double precision).
         """
-        return F64Type.get()
+        return F64Type.get() if width == 64 else F32Type.get()
 
     def getFloatAttr(self, type, value):
         """
-        Return an MLIR float attribute (double precision).
+        Return an MLIR float attribute (single or double precision).
```

Comment:
I guess someone should ask: there are no plans for half or quad precision, right?

Reply:
I think we should add them, but it is better to add it separately when needed - trying to not to grow this PR any bigger...
Will create and issue for that as part of https://github.com/NVIDIA/cuda-quantum/issues/1549